### PR TITLE
PBS Server Upgrade: Version 0.226.0

### PR DESCRIPTION
**Pull request by jorgeluisrocha**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1095,7 +1097,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("analytics.pubstack.buffers.count", 100)
 	v.SetDefault("analytics.pubstack.buffers.timeout", "900s")
 	v.SetDefault("amp_timeout_adjustment_ms", 0)
-	v.BindEnv("gdpr.default_value")
+	v.SetDefault("gdpr.default_value", 0)
```
Comment (karimMourra) [inline config/config.go:1100]: some config changes are missing see comparison between our fork and the equivalent tag
https://github.com/jwplayer/prebid-server/compare/v0.178.0...jwplayer:prebid-server:pbs_v0.178.0

File: adapters/jwplayer/jwplayer.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+	if source == nil {
+		return
+	}
+	source.Ext = nil
+}
+
+func (a *Adapter) clearPublisherSChain26(source *openrtb2.Source) {
+	if source == nil {
+		return
+	}
```
Comment (karimMourra) [inline adapters/jwplayer/jwplayer.go]: this check is unnecessary; without a source you can't have a 2.6 schain so this code would never run.

File: adapters/jwplayer/jwplayer_test.go [hunk 3]
```diff
@@ -0,0 +1,802 @@
...
+		Imp: []openrtb2.Imp{{
+			ID:  "test_imp_id",
+			Ext: json.RawMessage(`{"bidder":{"placementId": "1"}}`),
+		}},
+		Site: &openrtb2.Site{
+			Publisher: &openrtb2.Publisher{
+				Ext: json.RawMessage(`{"jwplayer":{"publisherId": "testPublisherId"}}`),
+			},
+		},
+		Source: &openrtb2.Source{},
```
Comment (karimMourra) [inline adapters/jwplayer/jwplayer_test.go:774]: this unit test changed at this line. The test should not change. We need to have test for ortb 2.5 and tests for ortb 2.6. The old tests should not be modified.

File: adapters/jwplayer/jwplayer_test.go [hunk 4]
```diff
@@ -0,0 +1,802 @@
...
+				Ext: json.RawMessage((`{"jwplayer":{"publisherId":"testPublisherId"}}`)),
+			},
+		},
+		Device: &openrtb2.Device{},
+		Ext:    json.RawMessage((`{"schain":{"complete":1,"nodes":[{"asi":"jwplayer.com","sid":"testPublisherId","rid":"test_id","hp":1}],"ver":"1.0"}}`)),
+	}
+	assert.Equal(t, expectedRequest, bidRequest)
+}
+
+func TestAppendingToExistingSchain(t *testing.T) {
```
Comment (karimMourra) [inline adapters/jwplayer/jwplayer_test.go]: THis test was changed. The test should not change. We need to have tests for ortb 2.5 and tests for ortb 2.6. The old tests should not be modified.

File: adapters/jwplayer/ortbHelpers.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+		return nil
+	}
+
+	if source.SChain == nil {
+		return nil
+	}
+
+	schain := *source.SChain
+
+	return &schain
```
Comment (karimMourra) [inline adapters/jwplayer/ortbHelpers.go]: why not just return source.schain ? You're getting the pointer and then returning the value anyways. It doesn't seem necessary the back and forth

File: adapters/smartadserver/smartadservertest/supplemental/request-site-recreated.json [hunk 6]
```diff
@@ -27,7 +27,7 @@
   "httpCalls": [
     {
       "expectedRequest": {
-        "uri": "https://ssb.smartadserver.com/api/bid?callerId=5",
+        "uri": "https://ssb.smartadserver.com/api/bid?callerId=50",
```
Comment (karimMourra) [inline adapters/smartadserver/smartadservertest/supplemental/request-site-recreated.json:30]: are these changes necessary ?

**Review by karimMourra — CHANGES_REQUESTED** [review]

File: adapters/smartadserver/smartadservertest/supplemental/request-site-recreated.json (hunk 6 above)
Comment (jorgeluisrocha) [reply adapters/smartadserver/smartadservertest/supplemental/request-site-recreated.json:30]: yes, this was explicitly requested by smart adserver.

File: adapters/jwplayer/ortbHelpers.go (hunk 5 above)
Comment (jorgeluisrocha) [reply adapters/jwplayer/ortbHelpers.go]: sure, can be fixed.

File: adapters/jwplayer/jwplayer.go (hunk 2 above)
Comment (jorgeluisrocha) [reply adapters/jwplayer/jwplayer.go]: will remove, this is a holdover from the old version of the clearPublisher function.

File: adapters/jwplayer/jwplayer_test.go (hunk 3 above)
Comment (jorgeluisrocha) [reply adapters/jwplayer/jwplayer_test.go:774]: modified the unit test to accommodate the changes for 2.6, but did not put back the old test!

File: usersync/cookie_test.go [hunk 7]
```diff
@@ -359,7 +359,8 @@ func TestTrimCookiesClosestExpirationDates(t *testing.T) {
 	testCases := []aTest{
 		{maxCookieSize: 2000, expKeys: []string{"k1", "k2", "k3", "k4", "k5", "k6", "k7"}}, //1 don't trim, set
 		{maxCookieSize: 0, expKeys: []string{"k1", "k2", "k3", "k4", "k5", "k6", "k7"}},    //2 unlimited size: don't trim, set
-		{maxCookieSize: 800, expKeys: []string{"k1", "k5", "k4", "k3"}},                    //3 trim to size and set
+		// failing doing to size differences across platforms
```
Comment (dshore) [inline usersync/cookie_test.go:362]: Is this still necessary?  Maybe check if this flakiness was addressed since we forked?

**Review by dshore — CHANGES_REQUESTED** [review]

File: usersync/cookie_test.go (hunk 7 above)
Comment (karimMourra) [reply usersync/cookie_test.go:362]: @jorgeluisrocha did you check this ?

File: config/config.go (hunk 1 above)
Comment (karimMourra) [reply config/config.go:1100]: @jorgeluisrocha did you address this ?

Comment (jorgeluisrocha) [reply config/config.go:1100]: yes

File: usersync/cookie_test.go (hunk 7 above)
Comment (jorgeluisrocha) [reply usersync/cookie_test.go:362]: yes! its still flaky.
